posthog/queries/base.py

```diff
@@ -164,10 +164,103 @@ def match_property(property: Property, override_property_values: Dict[str, Any])
+
+    # short circuit query if key exists in override_property_values
+    if property.key in override_property_values and property.operator != "is_not_set":
+        # if match found, do nothing to Q
+        # if not found, return empty Q
+        if not match_property(property, override_property_values):
+            return Q(pk__isnull=True)
+        else:
+            return Q()
+
+    # if no override matches, return a true Q object
+
+    column = "group_properties" if property.type == "group" else "properties"
+
+    if property.operator == "is_not":
+        return Q(~lookup_q(f"{column}__{property.key}", value) | ~Q(**{f"{column}__has_key": property.key}))
+    if property.operator == "is_set":
+        return Q(**{f"{column}__{property.key}__isnull": False})
+    if property.operator == "is_not_set":
+        return Q(**{f"{column}__{property.key}__isnull": True})
+    if property.operator in ("regex", "not_regex") and not is_valid_regex(value):
+        # Return no data for invalid regexes
+        return Q(pk=-1)
+    if isinstance(property.operator, str) and property.operator.startswith("not_"):
+        return Q(
+            ~Q(**{f"{column}__{property.key}__{property.operator[4:]}": value})
+            | ~Q(**{f"{column}__has_key": property.key})
+            | Q(**{f"{column}__{property.key}": None})
+        )
+
+    if property.operator in ("is_date_after", "is_date_before"):
+        effective_operator = "gt" if property.operator == "is_date_after" else "lt"
+        return Q(**{f"{column}__{property.key}__{effective_operator}": value})
+
+    if property.operator == "exact" or property.operator is None:
+        return lookup_q(f"{column}__{property.key}", value)
+    else:
+        assert not isinstance(value, list)
+        return Q(**{f"{column}__{property.key}__{property.operator}": value})
+
+
+def property_group_to_Q(property_group: PropertyGroup, override_property_values: Dict[str, Any] = {}) -> Q:
+
+    filters = Q()
+
+    if not property_group or len(property_group.values) == 0:
+        return filters
+
+    if isinstance(property_group.values[0], PropertyGroup):
+        for group in property_group.values:
+            group_filter = property_group_to_Q(cast(PropertyGroup, group), override_property_values)
+            if property_group.type == PropertyOperatorType.OR:
+                filters |= group_filter
+            else:
+                filters &= group_filter
+    else:
+        for property in property_group.values:
+            property_filter = property_to_Q(cast(Property, property), override_property_values)
+            if property_group.type == PropertyOperatorType.OR:
+                filters |= property_filter
+            else:
+                filters &= property_filter
+
+    return filters
+
+
 def properties_to_Q(
     properties: List[Property],
-    team_id: int,
-    is_direct_query: bool = False,
     override_property_values: Dict[str, Any] = {},
 ) -> Q:
     """
```

Comment:
There is a comment here that doesn't apply anymore.

Also, is it necessary to keep this function? Seems like it's just a wrapping `property_group_to_Q`. (Maybe there's too much refactoring for this PR?

Reply:
It's here for now because it's still used in `flag_matching`, didn't want to touch that too, but you're right, could convert regular flag matching to a property group as well

Reply:
regarding too much refactoring.. I don't think so. This is because if I didn't move `property_to_Q` out, every import there needed to be inline, because everything would be a circular import, which was creating a lot of pains: this function was simply in the wrong place.

Moving it out allowed me to massively simplify things, and move the overrides right into it as well, which would've been a lot nastier if I kept it where it was (since that implied passing overriden properties to a property instance so `property_to_Q(overrides)` could parse them. 😓